pages/Transaction.tsx

```diff
@@ -2,90 +2,129 @@ import { Nip47Transaction } from "@getalby/sdk/dist/NWCClient";
 import dayjs from "dayjs";
 import { useLocalSearchParams } from "expo-router";
 import React from "react";
-import { View, TouchableOpacity } from "react-native";
+import { View, TouchableOpacity, ScrollView } from "react-native";
 import Screen from "~/components/Screen";
 import { MoveDownLeft, MoveUpRight } from "~/components/Icons";
 import { Text } from "~/components/ui/text";
 import { useGetFiatAmount } from "~/hooks/useGetFiatAmount";
 import { cn } from "~/lib/utils";
 import * as Clipboard from "expo-clipboard";
 import Toast from "react-native-toast-message";
+import Hex from "crypto-js/enc-hex";
+import Utf8 from "crypto-js/enc-utf8";
+
+type TLVRecord = {
+  type: number;
+  value: string;
+};
+
+type Boostagram = {
+  app_name: string;
+  name: string;
+  podcast: string;
+  url: string;
+  episode?: string;
+  itemID?: string;
+  ts?: string;
+  message?: string;
+  sender_id: string;
+  sender_name: string;
+  time: string;
+  action: string;
+  value_msat_total: number;
+}
 
 export function Transaction() {
   const { transactionJSON } = useLocalSearchParams() as unknown as {
     transactionJSON: string;
   };
   const transaction: Nip47Transaction = JSON.parse(transactionJSON);
   const getFiatAmount = useGetFiatAmount();
+  const rawMetadata = transaction.metadata;
```

Comment:
This part could potentially be memoized. Otherwise the records are parsed with every render...